terraform/modules/app-ecs-albs/main.tf

```diff
@@ -279,14 +287,97 @@ resource "aws_route53_record" "alerts_private_record" {
+  domain_name       = "prom.${local.subdomain}"
+  validation_method = "DNS"
+
+  subject_alternative_names = ["${aws_route53_record.prom_alias.*.fqdn}"]
+
+  lifecycle {
+    # We can't destroy a certificate that's in use, and we can't stop
+    # using it until the new one is ready.  Hence
+    # create_before_destroy here.
+    create_before_destroy = true
+  }
+}
+
+resource "aws_route53_record" "prometheus_cert_validation" {
+  # Count matches the domain_name plus each `subject_alternative_domain`
+  count = "${1 + local.prom_records_count}"
+
+  name       = "${lookup(aws_acm_certificate.prometheus_cert.domain_validation_options[count.index], "resource_record_name")}"
+  type       = "${lookup(aws_acm_certificate.prometheus_cert.domain_validation_options[count.index], "resource_record_type")}"
+  zone_id    = "${var.zone_id}"
+  records    = ["${lookup(aws_acm_certificate.prometheus_cert.domain_validation_options[count.index], "resource_record_value")}"]
+  ttl        = 60
+  depends_on = ["aws_acm_certificate.prometheus_cert"]
+}
+
+resource "aws_acm_certificate_validation" "prometheus_cert" {
+  certificate_arn         = "${aws_acm_certificate.prometheus_cert.arn}"
+  validation_record_fqdns = ["${aws_route53_record.prometheus_cert_validation.*.fqdn}"]
+}
+
+resource "aws_route53_record" "prom_alias" {
+  count = "${local.prom_records_count}"
+
+  zone_id = "${var.zone_id}"
+  name    = "prom-${count.index + 1}"
+  type    = "A"
+
+  alias {
+    name                   = "${aws_lb.prometheus_alb.dns_name}"
+    zone_id                = "${aws_lb.prometheus_alb.zone_id}"
+    evaluate_target_health = false
+  }
+}
+
+resource "aws_security_group" "prometheus_alb" {
+  name        = "${var.stack_name}-prometheus-alb"
+  description = "Access to prometheus ALB (${var.stack_name})"
+  vpc_id      = "${local.vpc_id}"
+}
+
+# Prometheus is fronted by an nginx which controls access to
+# either approved IP addresses, or users with basic auth creds
+# so we need to allow all IPs here
+resource "aws_security_group_rule" "prom_allow_http" {
+  security_group_id = "${aws_security_group.prometheus_alb.id}"
+  type              = "ingress"
+  from_port         = 80
+  to_port           = 80
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "prom_allow_https" {
+  security_group_id = "${aws_security_group.prometheus_alb.id}"
+  type              = "ingress"
+  from_port         = 443
+  to_port           = 443
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+}
+
 resource "aws_lb" "prometheus_alb" {
   name               = "${var.stack_name}-prometheus-alb"
   internal           = false
   load_balancer_type = "application"
-  security_groups    = ["${data.terraform_remote_state.infra_security_groups.monitoring_external_sg_id}"]
+
+  security_groups = [
+    "${data.terraform_remote_state.infra_security_groups.monitoring_external_sg_id}",
```

Comment:
We still need the monitoring_external_sg to be attached to the ALB?

Reply:
Ah I see your commit message saying this can later be removed 👍

Reply:
I think I misread your commit message and got confused about which security group it was referring to. I think we do need to remove this now (or when we do the traffic swap over) though as it allows all public traffic to alertmanager. I can see no benefit of keeping it at the moment however so will remove it now.

Reply:
No I got confused again, this is fine for the moment as this is the prometheus ALB whilst I was thinking of the alertmanager ALB.